sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncCreateAccountFragment.kt

```diff
@@ -0,0 +1,121 @@
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.Command
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.Command.AbortFlow
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.Command.Error
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.Command.FinishSetupFlow
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.ViewMode.CreatingAccount
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.ViewMode.SignedIn
+import com.duckduckgo.sync.impl.ui.setup.SyncCreateAccountViewModel.ViewState
+import javax.inject.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(FragmentScope::class)
+class SyncCreateAccountFragment : DuckDuckGoFragment(R.layout.fragment_sync_setup) {
+    @Inject
+    lateinit var viewModelFactory: FragmentViewModelFactory
+
+    private val binding: FragmentSyncSetupBinding by viewBinding()
+
+    private val viewModel: SyncCreateAccountViewModel by lazy {
+        ViewModelProvider(this, viewModelFactory)[SyncCreateAccountViewModel::class.java]
+    }
+
+    private val listener: SetupFlowListener?
+        get() = activity as? SetupFlowListener
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiEvents()
+        configureListeners()
+    }
+
+    private fun configureListeners() {
+        binding.closeIcon.setOnClickListener {
+            viewModel.onCloseClicked()
+        }
+        binding.footerButton.setOnClickListener {
+            viewModel.onNextClicked()
+        }
+    }
+
+    private fun observeUiEvents() {
+        viewModel
+            .viewState()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.CREATED)
+            .onEach { viewState -> renderViewState(viewState) }
+            .launchIn(lifecycleScope)
+
+        viewModel
+            .commands()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach { processCommand(it) }
+            .launchIn(lifecycleScope)
+    }
+
+    private fun renderViewState(viewState: ViewState) {
+        when (viewState.viewMode) {
+            is SignedIn -> {
+                binding.footerButton.isEnabled = true
+                binding.footerButton.setOnClickListener {
+                    listener?.launchFinishSetupFlow()
+                }
+            }
+
+            CreatingAccount -> {
+                binding.footerButton.isEnabled = false
+            }
+        }
+    }
+
+    private fun processCommand(it: Command) {
+        when (it) {
+            AbortFlow -> {
+                requireActivity().setResult(Activity.RESULT_CANCELED)
+                requireActivity().finish()
+            }
+
+            FinishSetupFlow -> listener?.launchFinishSetupFlow()
+            Error -> {
```

Comment:
Similar question. Do we need to do anything here, if not we can leave a comment to clarify is intended